src/NHibernate/Transaction/AdoNetWithDistributedTransactionFactory.cs

```diff
@@ -104,6 +104,79 @@ public DistributedTransactionContext(ISessionImplementor sessionImplementor, Sys
 				IsInActiveTransaction = true;
 			}
 
+			#region IPromotableSinglePhaseNotification Members
+
+			void IPromotableSinglePhaseNotification.Initialize()
```

Comment:
It does not look to me as a valid place for flushing before commit. It flushes too early here, right at the beginning of the transaction I believe, not "just before actual commit". Anyway currently I think this code here has no actual usage (never called).